File: level2/ex20.js
function solution(n, k) {
  // 소수인지 판별하는 함수
  function isPrime(number) {
    // 숫자가 아닌 경우 소수가 아니다.
    if (isNaN(number)) return false;

    // 0과 1은 소수가 아니다.
    if (number < 2) return false;
    // 2보다 큰 짝수는 소수가 아니다.
    if (number > 2 && number % 2 === 0) return false;

    for (let i = 2; i <= Math.sqrt(number); i++) {
      // 약수가 존재하는 경우
      if (number % i === 0) return false;
    }

    return true;
  }

  // 10진수를 k진수로 변환하는 함수
  function changeBase(n, k) {
    let number = '';

    while (n > 0) {
      number += n % k;
      n = parseInt(n / k);
    }

    return number.split('').reverse().join('');
  }

  // 진법 변환 후 0을 기준으로 쪼개기
  const arr = changeBase(n, k).split(0);

  // 소수의 개수 세기
  let count = 0;

  for (let i = 0; i < arr.length; i++) {
    // 문자열을 정수로 변환
    const number = parseInt(arr[i]);

    // 소수인지 판별하여 개수 증가
    if (isPrime(number)) count++;
  }

  return count;
}

File: level2/ex20.py
def solution(n, k):
    # 10진수 정수를 k진수로 변환하는 함수
    def convert_base(n, k):
        # k진수 문자열
        n_k = ''
        
        while n > 0:
            mod = n % k
            n_k += str(mod)
            n = int(n / k)
            
        return n_k[::-1]

    # 소수 판별 함수
    def isprime(n):
        if n < 2:
            return False
        for i in range(2, int(n ** 0.5) + 1):
            if n % i == 0:
                return False
        return True
    
    # 조건에 맞는 소수의 개수
    count = 0
    
    # 10진수 n을 k진수로 변환한 결과 문자열
    n_k = convert_base(n, k)
    
    # 0을 기준으로 split한 결과 리스트
    numbers = n_k.split('0')
    
    for number in numbers:
        if number.isdigit() and isprime(int(number)):
            count += 1

    return count